libcfnet/tls_generic.c

```diff
@@ -724,7 +724,20 @@ int TLSRecv(SSL *ssl, char *buffer, int toget)
     assert(toget < CF_BUFSIZE);
     assert_SSLIsBlocking(ssl);
 
-    int received = SSL_read(ssl, buffer, toget);
+    int received,tries = 0,code=0;
+    do
```

Comment:
Why not use SSL_MODE_AUTO_RETRY? Or maybe this is another failure case you are working on?

Reply:
I think it's better to have the control over the retries.